helix-term/src/ui/editor.rs

```diff
@@ -442,17 +442,15 @@ impl EditorView {
             }
 
             let starting_indent =
-                (offset.col / tab_width) as u16 + config.indent_guides.skip_levels;
-            // TODO: limit to a max indent level too. It doesn't cause visual artifacts but it would avoid some
-            // extra loops if the code is deeply nested.
+                (offset.col / tab_width) + config.indent_guides.skip_levels as usize;
 
-            for i in starting_indent..(indent_level / tab_width as u16) {
-                surface.set_string(
-                    viewport.x + (i * tab_width as u16) - offset.col as u16,
-                    viewport.y + line,
-                    &indent_guide_char,
-                    indent_guide_style,
-                );
+            for i in starting_indent..(indent_level / tab_width) {
+                let x = (viewport.x as usize + (i * tab_width) - offset.col) as u16;
+                let y = viewport.y + line;
+                if !surface.in_bounds(x, y) {
+                    break;
+                }
```

Comment:
This is an improvement but we still do a bunch of calculation for each `i` that's out of bounds.

Could we precompute the `starting_indent..end` that are in bounds before the loop?

Reply:
It's ok to leave this as a follow-up, just add a TODO item here